### ISSUE TITLE
TST: Eliminate usage of check_less_precise keyword

### ISSUE THREAD

**Issue by jdmcbr** [geopandas/geopandas#1754]
The `check_less_precise` keyword in the `assert_frame_equal` testing method is deprecated as of pandas 1.1.0 (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.testing.assert_frame_equal.html). We should eliminate our usage of this keyword in favor of `atol` and `rtol`. See also https://github.com/pandas-dev/pandas/pull/30562.

**Comment by johnflavin** [comment]
I do support this. I have recently written tests where I wanted to use `atol` to control the precision when comparing two geodataframes, but it doesn't exist on `assert_geodataframe_equal`. I had to split into two assertions: `assert_frame_equal` on the non-geometry columns and `assert_geoseries_equal` on the geometry column.

But I would rephrase this enhancement. In pandas testing, `check_less_precise` was about numerical precision in any column and `atol` and `rtol` are replacements for that. But here `check_less_precise` controls how to compare the geometries (using `geom_equals` or `geom_almost_equals`) and there is no kwarg to control non-geometry numerical precision. We can keep `check_less_precise` and add `atol` and `rtol`.
